File: apps/backend/app/models/account.py
from typing import Any, Literal

from pydantic import BaseModel, Field


class ProfileUpdate(BaseModel):
    display_name: str = Field(min_length=1, max_length=100)


class Profile(BaseModel):
    id: str
    display_name: str
    created_at: str
    updated_at: str


class AccountOverview(BaseModel):
    id: str
    email: str | None
    created_at: str | None
    last_sign_in_at: str | None
    profile: Profile


class AccountDeletion(BaseModel):
    password: str = Field(min_length=1, max_length=1000)
    confirmation: Literal["DELETE"]


class DataExport(BaseModel):
    format_version: Literal[1] = 1
    exported_at: str
    account: dict[str, Any]
    profile: dict[str, Any] | None
    decisions: list[dict[str, Any]]
    decision_options: list[dict[str, Any]]
    decision_messages: list[dict[str, Any]]
    evaluations: list[dict[str, Any]]

File: apps/frontend/src/lib/api/account.ts
import { authenticatedFetch } from "./decisions";

export type AccountOverview = {
  id: string;
  email: string | null;
  created_at: string | null;
  last_sign_in_at: string | null;
  profile: {
    id: string;
    display_name: string;
    created_at: string;
    updated_at: string;
  };
};

export function getAccount() {
  return authenticatedFetch<AccountOverview>("/account", undefined, "/settings");
}

export function updateProfile(displayName: string) {
  return authenticatedFetch<AccountOverview["profile"]>(
    "/account/profile",
    { method: "PATCH", body: JSON.stringify({ display_name: displayName }) },
    "/settings",
  );
}

export function getAccountExport() {
  return authenticatedFetch<Record<string, unknown>>("/account/export", undefined, "/settings");
}

export function deleteAccount(password: string, confirmation: string) {
  return authenticatedFetch<void>(
    "/account",
    { method: "DELETE", body: JSON.stringify({ password, confirmation }) },
    "/settings",
  );
}
